### DISCUSSION TITLE
vercel.json - redirect patterns that match optional trailing slashes

### DISCUSSION THREAD

**Discussion by dabeeeenster** [vercel/community#180] [Help] [answered]
Hi,

I cant find any examples of this. I'm trying to set up redirects in my `vercel.json` file that match on urls both with and without a trailing slash in the `source` but I cant find out the right format. Does anyone know what I should be using? 

e.g. I want to do something like:

```
        {
            "source": "/legal/privacy-policy*",
            "destination": "/privacy-policy/"
        },
```

But vercel complains about the format of the file with the above

**Comment by Crizzooo** [comment]
Did you ever figure this out @dabeeeenster ?

**Reply by dabeeeenster** [reply]
Nope :(

**Comment by leerob** [comment, marked as answer]
For complete control over redirection logic and path matching, you can use Middleware. 

https://vercel.com/docs/concepts/functions/edge-functions#middleware

**Reply by Crizzooo** [reply]
> For complete control over redirection logic and path matching, you can use Middleware. 
> 
> https://vercel.com/docs/concepts/functions/edge-functions#middleware

@leerob i dont believe edge functions are possible with a gatsby.js codebase? We might have to retire gatsby for next.js to use middleware in this way

**Reply by leerob** [reply]
Yes, you can use Middleware with Gatsby! We are working to add more examples here. Place a `_middleware.js` file in your root directory.
